### PR TITLE
Comparisons, Julia/BioJulia installation sections + devcontainer

### LINKED ISSUE #2: Create a dev container definition
**Is your feature request related to a problem? Please describe.**
Ideally, users needn't worry about setting up a development environment when contributing. Some would prefer having an automated or easier process, such as how Docker templates can be used as recipes to [setup a local dev environment](https://blog.atulr.com/docker-local-environment/). 

**Describe the solution you'd like**
"Dev container" definitions extend the idea of containers for development by providing a JSON recipe to mount custom containers on VSCode and other IDEs. Thus, users may simply generate dev environments from a given _devcontainer.json_ file provided in the repo. 

BioJuliaDocs is a community-driven intiative, thus we would like to avoid build issues. Setting it up locally is, at the time of writing, straightfoward enough (install Julia, clone the repo, `julia docs/make.jl`), therefore the definition should be simple. This setup would provide good practice for extending dev containers to other BioJulia repos.

**Describe alternatives you've considered**
Provide an instructions list detailing how to install/run Julia and BioJuliaDocs locally.

**Additional context**
Relevant documentation for [dev containers](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers) and [Codespaces](https://docs.github.com/en/codespaces) .

### PR DESCRIPTION
**Pull request by M-PERSIC**
- Add scikit-bio, BioPerl, BioRuby, and BioJava comparison sections
- Finish julia/biojulia installation sections (have some more ideas for showing off vscode commands however it is ready for MVP)
- Initial skeleton section template for case scenarios (day in the life of a teacher, researcher, etc. and how they use Julia)
- Add devcontainer schema 

closes #2

### CODE REVIEW

**Comment by M-PERSIC** [comment]
~Ah one moment I found something interesting!~ All good

**Comment by kescobo** [comment]
Preview build: https://biojulia.dev/BioJuliaDocs/previews/PR9/

There's some way to get this automatically posted to the thread or in the merge checks when it's build, but I can't remember.

File: docs/src/getting_started/toolsinstallation.md [hunk 1]
```diff
@@ -1 +1,110 @@
...
+  and the editor by entering `julia> edit("<name of file>")`. You can jump right back into the Julia session once you're 
+  finished!
+- You can get a list of matches to a partially written word by entering the TAB key, which also serves to render an
+  Unicode symbol in the REPL (full list available in the [Unicode Input](https://docs.julialang.org/en/v1/manual/unicode-input/) section of the Julia manual): 
+  ```julia
+  julia> pri[TAB]
+  primitive type  print           println         printstyled
+  julia> \pi[TAB] # renders π
+  ```
+- You can run a standalone script by entering `include("<name of script>")` in Julian mode. For greater control, the
```
Comment (kescobo) [inline docs/src/getting_started/toolsinstallation.md:53]:
````suggestion
  Unicode symbol in the REPL (full list available in the [Unicode Input](https://docs.julialang.org/en/v1/manual/unicode-input/) section of the Julia manual): 

    ```julia
  julia> pri[TAB]
  primitive type  print           println         printstyled
  julia> \pi[TAB] # renders π
  ```

- You can run a standalone script by entering `include("<name of script>")` in Julian mode. For greater control, the 
````

More new-lines between things, in general

File: docs/src/getting_started/toolsinstallation.md [hunk 2]
```diff
@@ -1 +1,110 @@
...
+  ```julia
+  julia> pri[TAB]
+  primitive type  print           println         printstyled
+  julia> \pi[TAB] # renders π
+  ```
+- You can run a standalone script by entering `include("<name of script>")` in Julian mode. For greater control, the 
+  equivalent of Python3's `if __name__ == __main__` condition would be `if abspath(<name of program file>) == @__FILE__`
+
+!!! warning
+    One common footgun with the Julia REPL is that there are currently certain code changes that cannot be performed without
```
Comment (kescobo) [inline docs/src/getting_started/toolsinstallation.md:57]: Though "footgun" is a great phrase, it's probably less accessible for non-native speakers or folks new to development.

**Comment by M-PERSIC** [comment]
If it's a GitHub action, it should be easy to add. I added a note to the README on how to deploy locally (open up the website in localhost) so anyone can see their changes before submitting a pr

File: docs/src/getting_started/toolsinstallation.md [hunk 3]
```diff
@@ -1 +1,110 @@
...
+    @ REPL[2]:1
+    ```
+
+To learn more, see the [The Julia REPL](https://docs.julialang.org/en/v1/stdlib/REPL/) section in the Julia manual.
+
+## Visual Studio Code extension
+
+The Julia organization officially maintains the [Julia vscode extension](https://www.julia-vscode.org/), which provides
+a complete IDE solution for developing in Julia. This boasts a number of advantages compared to a REPL solution:
+- Built-in syntax highlighting and code completion
```
Comment (kescobo) [inline docs/src/getting_started/toolsinstallation.md:82]:
```suggestion

- Built-in syntax highlighting and code completion
```

File: docs/src/getting_started/biojuliainstallation.md [hunk 4]
```diff
@@ -1 +1,85 @@
...
+    line can be copy/pasted as is into the REPL! See the [Julia REPL](@ref julia-repl) section for more.
+
+2. **Pkg mode**
+
+```julia
+# Enter ]
+]
+# @v1.x will show your current Julia version
+(@v1.9) pkg> add <name of package here>
+# Exit REPL via the Backspace key
```
Comment (kescobo) [inline docs/src/getting_started/biojuliainstallation.md:33]:
```suggestion
# Exit Pkg REPL via the Backspace key
```

File: docs/src/getting_started/biojuliainstallation.md [hunk 5]
```diff
@@ -1 +1,85 @@
...
+julia> using <name of package here>
+```
+
+!!! note
+    Just like with the previous method, you can copy/paste code with the `(@v1.x) pkg>` prompt included
+    and Julia will work correctly! You can even skip having to enter Pkg mode by just entering directly
+    `(@v1.x) pkg> add...`. You do not need to manually change the version number indicated in the
```
Comment (kescobo) [inline docs/src/getting_started/biojuliainstallation.md:40]: I actually didn't know this! Wild.

File: docs/src/getting_started/biojuliainstallation.md [hunk 6]
```diff
@@ -1 +1,85 @@
...
+    One excellent trick is that any Julia code with the `julia>` prompt included at the beginning of the
+    line can be copy/pasted as is into the REPL! See the [Julia REPL](@ref julia-repl) section for more.
+
+2. **Pkg mode**
+
+```julia
+# Enter ]
+]
+# @v1.x will show your current Julia version
+(@v1.9) pkg> add <name of package here>
```
Comment (kescobo) [inline docs/src/getting_started/biojuliainstallation.md:32]: Perhaps a note / warning about installing things in your global environment - suggest folks make project-specific environments

**Review by kescobo — APPROVED** [review]
Looking good!

File: docs/src/getting_started/toolsinstallation.md (hunk 2 above)
Comment (M-PERSIC) [reply docs/src/getting_started/toolsinstallation.md:57]: How about "a common gotcha"?

**Comment by M-PERSIC** [comment]
I'll make a quick second pr for some changes.

**Pull request merged**